### PR TITLE
[new release] flint (0.4.2)

### PR DESCRIPTION
**Pull request by bobot**
Stub of the C library Flint3

- Project page: <a href="https://github.com/bobot/ocaml-flint">https://github.com/bobot/ocaml-flint</a>

##### CHANGES:

  - Fix compilation (@bclement-ocp)

### CODE REVIEW

**Comment by jmid** [comment]
> There is not a lot of users of this library, so next version those deprecated package will be removed. [,,,]

OK, thanks for sharing!

> > Platform-wise it looks like conf-flint on Alpine, Centos, and FreeBSD could use a hand
> 
> Normally alpine should be fixed. On the other hand the package doesn't exists anymore in centos, and doesn't exists in freebsd. I don't want to forbid to install the package on this distribution since a user can install the C library manually.

Good point!

> Perhaps there is some `ci-dont-test: {os = freebsd}`?

We cannot prevent the CI run but we can silence the CI-error on FreeBSD: :shrug: 
```
x-ci-accept-failures: ["freebsd-14.3"]
```

> For the segfault on arm32 and ppc, I'm in the process of changing the way the binding is written. So it could disappear then. Should I do something for those platform in the mean time?

If the packages work reasonably on arm32 and ppc64 except for test suite corner cases, we could just disable runtest on those platforms:
```
    "@runtest" {with-test & arch != "arm32" & arch != "ppc64"}
```
Alternatively, we could just disable the packages entirely on the platforms to prevent potential users from chasing segfault ghosts: :ghost: 
```
available: arch != "arm32" & arch != "ppc64"
```

File: packages/conf-flint/conf-flint.3.0/opam [hunk 1]
```diff
@@ -22,12 +27,14 @@ depexts: [
...
   ["flint" "flint-devel"] {os-family = "fedora"}
   ["flint" "flint-devel"] {os-distribution = "ol"}
-  ["flint-dev"] {os-distribution = "alpine"}
+  ["flint-dev@testing"] {os-distribution = "alpine"}
   ["flint-devel"] {os-family = "opensuse"}
-  ["libflint-devel"] {os = "win32" & os-distribution = "cygwinports"}
+  ["libflint-devel"] {os = "win32" &
+                         (os-distribution = "cygwinports" |
+                          os-distribution = "cygwin") }
+  ["mingw-w64-flint"] {os = "win32" & os-distribution = "msys2" }
```
Comment (jmid) [inline packages/conf-flint/conf-flint.3.0/opam]: I can see your points:
- `flint` is [not available as a MinGW Cygwin package](https://cygwin.com/packages/package_list.html), so we instead just use the Cygwin package. Even though that may make CI happy (actually it still fails!) that's not the way to go, because the resulting executable will then require `cygwin1.dll` contrary to MinGW expectations
- the MSYS2 fix is fair enough. It is 64-bit only, but [MSYS2 doesn't offer a 32-bit package](https://packages.msys2.org/base/mingw-w64-flint)

File: packages/conf-flint/conf-flint.3.0/opam [hunk 2]
```diff
@@ -22,12 +27,14 @@ depexts: [
...
-  ["flint" "flint-devel"] {os-distribution = "centos"}
   ["flint" "flint-devel"] {os-family = "fedora"}
   ["flint" "flint-devel"] {os-distribution = "ol"}
-  ["flint-dev"] {os-distribution = "alpine"}
+  ["flint-dev@testing"] {os-distribution = "alpine"}
   ["flint-devel"] {os-family = "opensuse"}
-  ["libflint-devel"] {os = "win32" & os-distribution = "cygwinports"}
+  ["libflint-devel"] {os = "win32" &
+                         (os-distribution = "cygwinports" |
+                          os-distribution = "cygwin") }
```
Comment (jmid) [inline packages/conf-flint/conf-flint.3.0/opam]:
```suggestion
  ["libflint-devel"] {os = "win32" & os-distribution = "cygwinports" }
  ["libflint-devel"] {os = "cygwin" & os-distribution = "cygwin" }
```
I suggest the following instead:
- reverting the cygwinports case
- adding Cygwin support (pure Cygwin, not MinGW: https://github.com/ocaml/opam-repository/wiki/Depexts-os-distribution---os-family-values )
- not supporting MinGW under Cygwin (which we can't because it lacks the needed system package - and which fails in CI anyway)

File: packages/conf-flint/conf-flint.3.0/opam (hunk 1 above)
Comment (bobot) [reply packages/conf-flint/conf-flint.3.0/opam]: Yes the cygwin package is outdated: https://cygwin.com/packages/summary/flint-src.html . I hope someone will update it. The process to update the package was not straightforward at first glance and msys2 seems a lively alternative.
